include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -573,6 +574,48 @@ class __future : private std::tuple<_Args...>
     }
 };
 
+// A contract for a future class for reduce: <execution policy, sycl::event, USM host memory for the reduced value>
+template <typename _ExecutionPolicy, typename _Event, typename _Res>
+class __reduce_future
+{
+    _ExecutionPolicy __my_exec;
+    _Event __my_event;
+    ::std::shared_ptr<_Res> __my_res;
+
+  public:
+    __reduce_future(const _ExecutionPolicy& __exec, _Event __e, _Res __res)
+        : __my_exec(__exec), __my_event(__e), __my_res(::std::make_shared<_Res>(__res))
```

Comment:
Since `_Event __e` is passed by value in the constructor, in the initializer list it may be better to have `__my_event(::std::move(__e))` to avoid an unnecessary copy.

Reply:
Agreed. I will change that.